### ISSUE TITLE
Redundant modifier false positive for abstract classes in interfaces (6.9)

### ISSUE THREAD

**Issue by don-vip** [checkstyle/checkstyle#1833]
Hi,
The new check for redundant modifier in version 6.9 triggers one false positive in our project:

https://josm.openstreetmap.de/jenkins/job/JOSM/69/checkstyleResult/NORMAL/

"abstract" is not redundant => removing it causes a compile error.

```java
/**
 * https://github.com/checkstyle/checkstyle/issues/1833
 */
public interface Selector {

    void getRange();

    abstract class AbstractSelector implements Selector {

    }
}
```
```
$ java -jar checkstyle-6.9-all.jar -c /sun_checks.xml Selector.java
Selector.java:8:5: Redundant 'abstract' modifier.
```

**Comment by romani** [comment]
please provide code example and config to let us easily reproduce a problem and fix by UT.

you can use out CLI to reproduce a problem - http://checkstyle.sourceforge.net/cmdline.html#Download_and_Run

**Comment by don-vip** [comment]
Sure, here it is:
```java
/**
 * https://github.com/checkstyle/checkstyle/issues/1833
 */
public interface Selector {

    void getRange();

    abstract class AbstractSelector implements Selector {

    }
}
```
```
$ java -jar checkstyle-6.9-all.jar -c /sun_checks.xml Selector.java
Selector.java:8:5: Redundant 'abstract' modifier.
```

**Comment by romani** [comment]
classes inside of interfaces :) , that is definitely missed point in out UTs.
We need to also recheck all other modifiers of inner classes , example - http://stackoverflow.com/questions/2400828/inner-class-within-interface

**Comment by romani** [comment]
merged
